src/nvim/os/signal.c

```diff
@@ -165,8 +165,6 @@ static char *signal_name(int signum)
 // This function handles deadly signals.
 // It tries to preserve any swap files and exit properly.
 // (partly from Elvis).
-// NOTE: Avoid unsafe functions, such as allocating memory, they can result in
-// a deadlock.
 static void deadly_signal(int signum)
```

Comment:
Let's add a comment here:

```suggestion
// NOTE: this is scheduled on the event loop, _not_ called directly from a signal handler.
static void deadly_signal(int signum)
```